test/parallel/xyk-pallet.API.pool.soloToken.test.ts

```diff
@@ -0,0 +1,280 @@
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqId)),
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(token1))
+  );
+
+  const rewardsSolo = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  const rewardsPool = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId
+  );
+
+  expect(rewardsSolo.rewardsAlreadyClaimed).bnEqual(
+    rewardsPool.rewardsAlreadyClaimed
+  );
+});
+
+test("GIVEN pool and solo token AND both were created at the same time AND the activated amounts are similar AND then some new activation happens after one session and some deactivation THEN the poolRewards storage is the same", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(liqId, BN_BILLION)),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(liqId, BN_MILLION)),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_MILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.deactivateLiquidity(liqId, BN_MILLION)),
+    Sudo.sudoAs(testUser1, Xyk.deactivateLiquidity(token1, BN_MILLION))
+  );
+
+  const rewardsSolo = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  const rewardsPool = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId
+  );
+
+  expect(rewardsSolo.poolRatioAtLastCheckpoint).bnEqual(
+    rewardsPool.poolRatioAtLastCheckpoint
+  );
+  expect(rewardsSolo.rewardsNotYetClaimed).bnEqual(
+    rewardsPool.rewardsNotYetClaimed
+  );
+});
+
+test("GIVEN a solo token rewards setup, WHEN weight goes from 20 to 0 THEN no more rewards will be granted for new users or new activations", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  const rewardsBefore = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(token1.toNumber(), 0),
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitNewStakingRound();
+
+  const rewardsAfter = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    token1
+  );
+
+  expect(rewardsAfter.missingAtLastCheckpoint).bnEqual(
+    rewardsBefore.missingAtLastCheckpoint
+  );
+
+  await Sudo.batchAsSudoFinalized(Assets.promotePool(token1.toNumber(), 20));
+});
+
+test("GIVEN a solo token rewards setup WHEN the user activates or deactivates THEN MPL is also modified", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.activateLiquidity(token1, BN_BILLION))
+  );
+
+  await waitForRewards(testUser1, token1);
+
+  const rewardsBefore = await getMultiPurposeLiquidityStatus(
```

Comment:
minor: rename. This is MPL status, with nothing to do with rewards.

Reply:
renamed